packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -641,30 +641,39 @@ class _MonthPickerState extends State<MonthPicker> {
       height: _kMaxDayPickerHeight,
       child: new Stack(
         children: <Widget>[
-          new PageView.builder(
-            key: new ValueKey<DateTime>(widget.selectedDate),
-            controller: _dayPickerController,
-            scrollDirection: Axis.horizontal,
-            itemCount: _monthDelta(widget.firstDate, widget.lastDate) + 1,
-            itemBuilder: _buildItems,
-            onPageChanged: _handleMonthPageChanged,
+          new Semantics(
+            sortKey: _MonthPickerSortKey.calendar,
```

Comment:
In general keys are grouped by type, so using plain `OrdinalSortKey` would cause these nodes to be sorted together with potentially unrelated nodes that roll up under the same parent. That's probably not going to happen in the current implementation of the picker.